hydromt/models/_v1/model_root.py

```diff
@@ -0,0 +1,187 @@
+
+        In read/append mode a check is done if the root exists.
+        In write mode the required model folder structure is created.
+
+        Parameters
+        ----------
+        root : str, optional
+            path to model root
+        mode : {"r", "r+", "w"}, optional
+            read/append/write mode for model files
+        """
+        self.set_mode(mode)
+        # set path is to take care of cross platform paths
+        self._set_path(path)
+        if self.is_writing_mode():
+            if create_dirs:
+                self._setup_folder_structure()
+                self._setup_log_file_hanglers()
+        if self.is_reading_mode():
+            self._check_root_exists()
+
+    def _check_root_exists(self):
+        # check directory
+        if not isdir(self._path):
+            raise IOError(f'model root not found at "{self._path}"')
+
+    def _setup_folder_structure(self):
+        ignore_ext = [".log", ".yml"]
+        for name in self.folders:
+            dir_path = join(self._path, name)
+            if not isdir(dir_path):
+                makedirs(dir_path)
+                continue
+            # path already exists, check files
+            files = glob(join(dir_path, "*.*"))
+            files = [Path(file).suffix for file in files]
+            files = [ext for ext in files if ext not in ignore_ext]
+            if len(files) != 0:
+                if self.mode.is_override():
+                    self.logger.warning(
+                        "Model dir already exists and "
+                        f"files might be overwritten: {self._path}."
+                    )
+                else:
+                    msg = (
+                        "Model dir already exists and cannot be "
+                        + f"overwritten: {self._path}. Use 'mode=w+' to force "
+                        + "overwrite existing files."
+                    )
+                    self.logger.error(msg)
+                    raise IOError(msg)
+
+    def _setup_log_file_hanglers(self) -> None:
+        # remove old logging file handler and add new filehandler
+        # in root if it does not exist
+        has_log_file = False
+        log_level = 20  # default, but overwritten by the level of active loggers
+        for i, handeler in enumerate(self.logger.handlers):
+            # make the type checkers a little happier
+            handeler = cast(FileHandler, handeler)
+            log_level = handeler.level
+            if hasattr(handeler, "baseFilename"):
+                if dirname(handeler.baseFilename) != self._path:
+                    # remove handler and close file
+                    self.logger.handlers.pop(i).close()
+                else:
+                    has_log_file = True
+                break
+        if not has_log_file:
+            new_path = join(self._path, "hydromt.log")
+            add_filehandler(self.logger, new_path, log_level)
+
+    def _set_path(self, path: PathLike) -> None:
```

Comment:
I think we should have a public ``ModelRoot.set_path`` function. You could decide to still keep this function that deals with posix/windows path and return the path in correct format and setting the actual path property can be done in the ``set_path`` function.